include/fmt/ranges.h

```diff
@@ -660,7 +660,9 @@ struct formatter<join_view<It, Sentinel, Char>, Char> {
   template <typename FormatContext>
   auto format(view_ref& value, FormatContext& ctx) const
```

Comment:
Shall we get rid of rvalue reference as discussed in the issue? Maybe replace `view_ref` with

```
  using view = conditional_t<std::is_copy_constructible<It>::value,
                                 const join_view<It, Sentinel, Char>,
                                 join_view<It, Sentinel, Char>>;
```
and use `std::is_copy_constructible` in the iterator check?

Reply:
Sure thing, made this change